ui/src/app/index/registration/modal/modal.component.ts

```diff
@@ -149,4 +151,12 @@ export class RegistrationModalComponent implements OnInit {
     return link.replace("{language}", this.service.getDocsLang());
   }
 
+  protected togglePasswordVisibility() {
+    this.showPassword = !this.showPassword;
+  }
+
+  protected toggleConfirmPasswordVisibility() {
+    this.showConfirmPassword = !this.showConfirmPassword;
```

Comment:
so the thing is, you are repeating this kind of logic over and over, seems unnecessary